packages/shared/components/Select/types.ts

```diff
@@ -31,6 +35,7 @@ export type Props = {
   width?: string | number;
   menuPlacement?: string;
   components?: any;
+  customProps?: { [key: string]: any };
```

Comment:
`customProps?: Record<string, any>` looks a bit nicer for me, but as you prefer :)